core/src/main/java/org/elasticsearch/common/xcontent/json/JsonXContent.java

```diff
@@ -43,7 +43,6 @@ public static XContentBuilder contentBuilder() throws IOException {
 
     static {
         jsonFactory = new JsonFactory();
-        jsonFactory.configure(JsonParser.Feature.ALLOW_UNQUOTED_FIELD_NAMES, true);
```

Comment:
I think we should explicitly set this to `false` so if Jackson's defaults ever change, we won't be bitten by it.